src/sentry_singleton.cpp

```diff
@@ -457,6 +502,9 @@ Sentry::Sentry() {
 		return;
 	}
 
+	// Load the runtime configuration from the user's data directory.
+	runtime_config.load_file(OS::get_singleton()->get_user_data_dir() + "/sentry.dat");
```

Comment:
There is an option in the project settings to use a custom user dir, but I don't think it would be useful here. Could implement a CLI option.

Reply:
Not sure either of those would help with concurrent tests and unit tests. 
In the unit test case, you'd want to mock that and setup different behavior (what if it throws, or returns A or B etc) to verify how the code calling it behaves.

In case of integration tests you'd at least want to make sure that each test has a single/random file folder/path to work on. So that they don't interact with each other. And at the end of the test run the root directory where all the tests access under, would be deleted, for clean up.